commit e615dae2244101e844d0851f3d9c8c754ffaffe0
Author: John Arthorne <johna>
Date:   Sun Feb 10 15:51:51 2008 +0000

    Bug 218232 [prov] Engine needs to be a service

1	1	bundles/org.eclipse.equinox.p2.console/src/org/eclipse/equinox/internal/p2/console/ProvisioningHelper.java
1	1	bundles/org.eclipse.equinox.p2.director.app/src/org/eclipse/equinox/internal/p2/director/app/Application.java
2	2	bundles/org.eclipse.equinox.p2.director/src/org/eclipse/equinox/internal/p2/director/SimpleDirector.java
2	1	bundles/org.eclipse.equinox.p2.engine/src/org/eclipse/equinox/internal/p2/engine/EngineActivator.java
1	1	bundles/org.eclipse.equinox.p2.engine/src/org/eclipse/equinox/p2/engine/BeginOperationEvent.java
1	1	bundles/org.eclipse.equinox.p2.engine/src/org/eclipse/equinox/p2/engine/CommitOperationEvent.java
1	1	bundles/org.eclipse.equinox.p2.engine/src/org/eclipse/equinox/p2/engine/Engine.java
11	0	bundles/org.eclipse.equinox.p2.engine/src/org/eclipse/equinox/p2/engine/IEngine.java
1	1	bundles/org.eclipse.equinox.p2.engine/src/org/eclipse/equinox/p2/engine/RollbackOperationEvent.java
1	1	bundles/org.eclipse.equinox.p2.engine/src/org/eclipse/equinox/p2/engine/TransactionEvent.java
2	2	bundles/org.eclipse.equinox.p2.tests/src/org/eclipse/equinox/p2/tests/AbstractProvisioningTest.java
3	3	bundles/org.eclipse.equinox.p2.tests/src/org/eclipse/equinox/p2/tests/engine/EngineTest.java
3	3	bundles/org.eclipse.equinox.p2.tests/src/org/eclipse/equinox/p2/tests/engine/InstallRegistryTest.java
3	3	bundles/org.eclipse.equinox.p2.tests/src/org/eclipse/equinox/p2/tests/engine/PhaseTest.java
3	3	bundles/org.eclipse.equinox.p2.tests/src/org/eclipse/equinox/p2/tests/engine/TouchpointTest.java
diff --git a/bundles/org.eclipse.equinox.p2.console/src/org/eclipse/equinox/internal/p2/console/ProvisioningHelper.java b/bundles/org.eclipse.equinox.p2.console/src/org/eclipse/equinox/internal/p2/console/ProvisioningHelper.java
index 0a80a85..1080cc2 100644
--- a/bundles/org.eclipse.equinox.p2.console/src/org/eclipse/equinox/internal/p2/console/ProvisioningHelper.java
+++ b/bundles/org.eclipse.equinox.p2.console/src/org/eclipse/equinox/internal/p2/console/ProvisioningHelper.java
@@ -195,7 +195,7 @@ public class ProvisioningHelper {
 		if (planner == null)
 			throw new ProvisionException("No planner service found.");
 
-		Engine engine = (Engine) ServiceHelper.getService(Activator.getContext(), Engine.class.getName());
+		IEngine engine = (IEngine) ServiceHelper.getService(Activator.getContext(), IEngine.SERVICE_NAME);
 		if (engine == null)
 			throw new ProvisionException("No director service found.");
 		IInstallableUnit[] toInstall = (IInstallableUnit[]) units.toArray(IInstallableUnit.class);
diff --git a/bundles/org.eclipse.equinox.p2.director.app/src/org/eclipse/equinox/internal/p2/director/app/Application.java b/bundles/org.eclipse.equinox.p2.director.app/src/org/eclipse/equinox/internal/p2/director/app/Application.java
index 1629026..7fcf458 100644
--- a/bundles/org.eclipse.equinox.p2.director.app/src/org/eclipse/equinox/internal/p2/director/app/Application.java
+++ b/bundles/org.eclipse.equinox.p2.director.app/src/org/eclipse/equinox/internal/p2/director/app/Application.java
@@ -186,7 +186,7 @@ public class Application implements IApplication {
 		if (planner == null)
 			throw new RuntimeException(Messages.Missing_planner);
 
-		Engine engine = (Engine) ServiceHelper.getService(Activator.getContext(), Engine.class.getName());
+		IEngine engine = (IEngine) ServiceHelper.getService(Activator.getContext(), IEngine.SERVICE_NAME);
 		if (engine == null)
 			throw new RuntimeException(Messages.Missing_Engine);
 
diff --git a/bundles/org.eclipse.equinox.p2.director/src/org/eclipse/equinox/internal/p2/director/SimpleDirector.java b/bundles/org.eclipse.equinox.p2.director/src/org/eclipse/equinox/internal/p2/director/SimpleDirector.java
index db39ca1..1617365 100644
--- a/bundles/org.eclipse.equinox.p2.director/src/org/eclipse/equinox/internal/p2/director/SimpleDirector.java
+++ b/bundles/org.eclipse.equinox.p2.director/src/org/eclipse/equinox/internal/p2/director/SimpleDirector.java
@@ -23,12 +23,12 @@ public class SimpleDirector implements IDirector {
 	private static final String ROLLBACK_LOCATION = "rollback"; //$NON-NLS-1$
 	static final int PlanWork = 10;
 	static final int EngineWork = 100;
-	private Engine engine;
+	private IEngine engine;
 	private IPlanner planner;
 
 	public SimpleDirector() {
 		initializeRollbackRepository();
-		engine = (Engine) ServiceHelper.getService(DirectorActivator.context, Engine.class.getName());
+		engine = (IEngine) ServiceHelper.getService(DirectorActivator.context, IEngine.SERVICE_NAME);
 		if (engine == null)
 			throw new IllegalStateException("Provisioning engine is not registered"); //$NON-NLS-1$
 		planner = (IPlanner) ServiceHelper.getService(DirectorActivator.context, IPlanner.class.getName());
diff --git a/bundles/org.eclipse.equinox.p2.engine/src/org/eclipse/equinox/internal/p2/engine/EngineActivator.java b/bundles/org.eclipse.equinox.p2.engine/src/org/eclipse/equinox/internal/p2/engine/EngineActivator.java
index 1a1d90c..f63ca62 100644
--- a/bundles/org.eclipse.equinox.p2.engine/src/org/eclipse/equinox/internal/p2/engine/EngineActivator.java
+++ b/bundles/org.eclipse.equinox.p2.engine/src/org/eclipse/equinox/internal/p2/engine/EngineActivator.java
@@ -12,6 +12,7 @@ package org.eclipse.equinox.internal.p2.engine;
 
 import org.eclipse.equinox.p2.core.eventbus.ProvisioningEventBus;
 import org.eclipse.equinox.p2.engine.Engine;
+import org.eclipse.equinox.p2.engine.IEngine;
 import org.osgi.framework.*;
 import org.osgi.util.tracker.ServiceTracker;
 import org.osgi.util.tracker.ServiceTrackerCustomizer;
@@ -31,7 +32,7 @@ public class EngineActivator implements BundleActivator, ServiceTrackerCustomize
 	public Object addingService(ServiceReference reference) {
 		if (registration == null) {
 			ProvisioningEventBus eventBus = (ProvisioningEventBus) context.getService(reference);
-			registration = context.registerService(Engine.class.getName(), new Engine(eventBus), null);
+			registration = context.registerService(IEngine.SERVICE_NAME, new Engine(eventBus), null);
 			return eventBus;
 		}
 		return null;
diff --git a/bundles/org.eclipse.equinox.p2.engine/src/org/eclipse/equinox/p2/engine/BeginOperationEvent.java b/bundles/org.eclipse.equinox.p2.engine/src/org/eclipse/equinox/p2/engine/BeginOperationEvent.java
index 93ddc63..b0867ea 100644
--- a/bundles/org.eclipse.equinox.p2.engine/src/org/eclipse/equinox/p2/engine/BeginOperationEvent.java
+++ b/bundles/org.eclipse.equinox.p2.engine/src/org/eclipse/equinox/p2/engine/BeginOperationEvent.java
@@ -14,7 +14,7 @@ public class BeginOperationEvent extends TransactionEvent {
 
 	private static final long serialVersionUID = 6389318375739324865L;
 
-	public BeginOperationEvent(IProfile profile, PhaseSet phaseSet, InstallableUnitOperand[] deltas, Engine engine) {
+	public BeginOperationEvent(IProfile profile, PhaseSet phaseSet, InstallableUnitOperand[] deltas, IEngine engine) {
 		super(profile, phaseSet, deltas, engine);
 	}
 }
diff --git a/bundles/org.eclipse.equinox.p2.engine/src/org/eclipse/equinox/p2/engine/CommitOperationEvent.java b/bundles/org.eclipse.equinox.p2.engine/src/org/eclipse/equinox/p2/engine/CommitOperationEvent.java
index 2ed6245..fe3e93c 100644
--- a/bundles/org.eclipse.equinox.p2.engine/src/org/eclipse/equinox/p2/engine/CommitOperationEvent.java
+++ b/bundles/org.eclipse.equinox.p2.engine/src/org/eclipse/equinox/p2/engine/CommitOperationEvent.java
@@ -13,7 +13,7 @@ package org.eclipse.equinox.p2.engine;
 public class CommitOperationEvent extends TransactionEvent {
 	private static final long serialVersionUID = -523967775426133720L;
 
-	public CommitOperationEvent(IProfile profile, PhaseSet phaseSet, InstallableUnitOperand[] deltas, Engine engine) {
+	public CommitOperationEvent(IProfile profile, PhaseSet phaseSet, InstallableUnitOperand[] deltas, IEngine engine) {
 		super(profile, phaseSet, deltas, engine);
 	}
 
diff --git a/bundles/org.eclipse.equinox.p2.engine/src/org/eclipse/equinox/p2/engine/Engine.java b/bundles/org.eclipse.equinox.p2.engine/src/org/eclipse/equinox/p2/engine/Engine.java
index 60024ae..19747f2 100644
--- a/bundles/org.eclipse.equinox.p2.engine/src/org/eclipse/equinox/p2/engine/Engine.java
+++ b/bundles/org.eclipse.equinox.p2.engine/src/org/eclipse/equinox/p2/engine/Engine.java
@@ -17,7 +17,7 @@ import org.eclipse.equinox.internal.p2.core.helpers.ServiceHelper;
 import org.eclipse.equinox.internal.p2.engine.*;
 import org.eclipse.equinox.p2.core.eventbus.ProvisioningEventBus;
 
-public class Engine {
+public class Engine implements IEngine {
 
 	private final ProvisioningEventBus eventBus;
 	private List lockedProfiles = new ArrayList();
diff --git a/bundles/org.eclipse.equinox.p2.engine/src/org/eclipse/equinox/p2/engine/IEngine.java b/bundles/org.eclipse.equinox.p2.engine/src/org/eclipse/equinox/p2/engine/IEngine.java
new file mode 100644
index 0000000..f91ef36
--- /dev/null
+++ b/bundles/org.eclipse.equinox.p2.engine/src/org/eclipse/equinox/p2/engine/IEngine.java
@@ -0,0 +1,11 @@
+package org.eclipse.equinox.p2.engine;
+
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+
+public interface IEngine {
+	public static final String SERVICE_NAME = IEngine.class.getName();
+
+	public IStatus perform(IProfile profile, PhaseSet phaseSet, Operand[] operands, ProvisioningContext context, IProgressMonitor monitor);
+
+}
diff --git a/bundles/org.eclipse.equinox.p2.engine/src/org/eclipse/equinox/p2/engine/RollbackOperationEvent.java b/bundles/org.eclipse.equinox.p2.engine/src/org/eclipse/equinox/p2/engine/RollbackOperationEvent.java
index 387441e..f962c4e 100644
--- a/bundles/org.eclipse.equinox.p2.engine/src/org/eclipse/equinox/p2/engine/RollbackOperationEvent.java
+++ b/bundles/org.eclipse.equinox.p2.engine/src/org/eclipse/equinox/p2/engine/RollbackOperationEvent.java
@@ -17,7 +17,7 @@ public class RollbackOperationEvent extends TransactionEvent {
 	private static final long serialVersionUID = -2076492953949691215L;
 	private IStatus cause;
 
-	public RollbackOperationEvent(IProfile profile, PhaseSet phaseSet, InstallableUnitOperand[] deltas, Engine engine, IStatus cause) {
+	public RollbackOperationEvent(IProfile profile, PhaseSet phaseSet, InstallableUnitOperand[] deltas, IEngine engine, IStatus cause) {
 		super(profile, phaseSet, deltas, engine);
 		this.cause = cause;
 	}
diff --git a/bundles/org.eclipse.equinox.p2.engine/src/org/eclipse/equinox/p2/engine/TransactionEvent.java b/bundles/org.eclipse.equinox.p2.engine/src/org/eclipse/equinox/p2/engine/TransactionEvent.java
index cf8d3fa..8d891b5 100644
--- a/bundles/org.eclipse.equinox.p2.engine/src/org/eclipse/equinox/p2/engine/TransactionEvent.java
+++ b/bundles/org.eclipse.equinox.p2.engine/src/org/eclipse/equinox/p2/engine/TransactionEvent.java
@@ -18,7 +18,7 @@ public abstract class TransactionEvent extends EventObject {
 	protected PhaseSet phaseSet;
 	protected InstallableUnitOperand[] deltas;
 
-	public TransactionEvent(IProfile profile, PhaseSet phaseSet, InstallableUnitOperand[] deltas, Engine engine) {
+	public TransactionEvent(IProfile profile, PhaseSet phaseSet, InstallableUnitOperand[] deltas, IEngine engine) {
 		super(engine);
 		this.profile = profile;
 		this.phaseSet = phaseSet;
diff --git a/bundles/org.eclipse.equinox.p2.tests/src/org/eclipse/equinox/p2/tests/AbstractProvisioningTest.java b/bundles/org.eclipse.equinox.p2.tests/src/org/eclipse/equinox/p2/tests/AbstractProvisioningTest.java
index d905629..39cc935 100644
--- a/bundles/org.eclipse.equinox.p2.tests/src/org/eclipse/equinox/p2/tests/AbstractProvisioningTest.java
+++ b/bundles/org.eclipse.equinox.p2.tests/src/org/eclipse/equinox/p2/tests/AbstractProvisioningTest.java
@@ -219,8 +219,8 @@ public class AbstractProvisioningTest extends TestCase {
 		return createIU(name, version, null, requires, BUNDLE_CAPABILITY, NO_PROPERTIES, ECLIPSE_TOUCHPOINT, touchpointData, false);
 	}
 
-	public static Engine createEngine() {
-		return (Engine) ServiceHelper.getService(TestActivator.getContext(), Engine.class.getName());
+	public static IEngine createEngine() {
+		return (IEngine) ServiceHelper.getService(TestActivator.getContext(), IEngine.SERVICE_NAME);
 	}
 
 	/**
diff --git a/bundles/org.eclipse.equinox.p2.tests/src/org/eclipse/equinox/p2/tests/engine/EngineTest.java b/bundles/org.eclipse.equinox.p2.tests/src/org/eclipse/equinox/p2/tests/engine/EngineTest.java
index 677e47e..d8d2fde 100644
--- a/bundles/org.eclipse.equinox.p2.tests/src/org/eclipse/equinox/p2/tests/engine/EngineTest.java
+++ b/bundles/org.eclipse.equinox.p2.tests/src/org/eclipse/equinox/p2/tests/engine/EngineTest.java
@@ -32,7 +32,7 @@ import org.osgi.framework.Version;
  */
 public class EngineTest extends AbstractProvisioningTest {
 	private ServiceReference engineRef;
-	private Engine engine;
+	private IEngine engine;
 	private File testProvisioning;
 
 	public EngineTest(String name) {
@@ -61,8 +61,8 @@ public class EngineTest extends AbstractProvisioningTest {
 	}
 
 	protected void setUp() throws Exception {
-		engineRef = TestActivator.getContext().getServiceReference(Engine.class.getName());
-		engine = (Engine) TestActivator.getContext().getService(engineRef);
+		engineRef = TestActivator.getContext().getServiceReference(IEngine.SERVICE_NAME);
+		engine = (IEngine) TestActivator.getContext().getService(engineRef);
 	}
 
 	protected void tearDown() throws Exception {
diff --git a/bundles/org.eclipse.equinox.p2.tests/src/org/eclipse/equinox/p2/tests/engine/InstallRegistryTest.java b/bundles/org.eclipse.equinox.p2.tests/src/org/eclipse/equinox/p2/tests/engine/InstallRegistryTest.java
index 51a7c7e..a0e6949 100644
--- a/bundles/org.eclipse.equinox.p2.tests/src/org/eclipse/equinox/p2/tests/engine/InstallRegistryTest.java
+++ b/bundles/org.eclipse.equinox.p2.tests/src/org/eclipse/equinox/p2/tests/engine/InstallRegistryTest.java
@@ -29,7 +29,7 @@ public class InstallRegistryTest extends AbstractProvisioningTest {
 	private ServiceReference registryRef;
 	private IInstallRegistry registry;
 	private ServiceReference engineRef;
-	private Engine engine;
+	private IEngine engine;
 
 	public InstallRegistryTest(String name) {
 		super(name);
@@ -42,8 +42,8 @@ public class InstallRegistryTest extends AbstractProvisioningTest {
 	protected void setUp() throws Exception {
 		registryRef = TestActivator.getContext().getServiceReference(IInstallRegistry.class.getName());
 		registry = (IInstallRegistry) TestActivator.getContext().getService(registryRef);
-		engineRef = TestActivator.getContext().getServiceReference(Engine.class.getName());
-		engine = (Engine) TestActivator.getContext().getService(engineRef);
+		engineRef = TestActivator.getContext().getServiceReference(IEngine.SERVICE_NAME);
+		engine = (IEngine) TestActivator.getContext().getService(engineRef);
 	}
 
 	protected void tearDown() throws Exception {
diff --git a/bundles/org.eclipse.equinox.p2.tests/src/org/eclipse/equinox/p2/tests/engine/PhaseTest.java b/bundles/org.eclipse.equinox.p2.tests/src/org/eclipse/equinox/p2/tests/engine/PhaseTest.java
index 5b0622f..e524aa9 100644
--- a/bundles/org.eclipse.equinox.p2.tests/src/org/eclipse/equinox/p2/tests/engine/PhaseTest.java
+++ b/bundles/org.eclipse.equinox.p2.tests/src/org/eclipse/equinox/p2/tests/engine/PhaseTest.java
@@ -88,7 +88,7 @@ public class PhaseTest extends AbstractProvisioningTest {
 	}
 
 	private ServiceReference engineRef;
-	private Engine engine;
+	private IEngine engine;
 
 	public PhaseTest(String name) {
 		super(name);
@@ -99,8 +99,8 @@ public class PhaseTest extends AbstractProvisioningTest {
 	}
 
 	protected void setUp() throws Exception {
-		engineRef = TestActivator.getContext().getServiceReference(Engine.class.getName());
-		engine = (Engine) TestActivator.getContext().getService(engineRef);
+		engineRef = TestActivator.getContext().getServiceReference(IEngine.SERVICE_NAME);
+		engine = (IEngine) TestActivator.getContext().getService(engineRef);
 	}
 
 	protected void tearDown() throws Exception {
diff --git a/bundles/org.eclipse.equinox.p2.tests/src/org/eclipse/equinox/p2/tests/engine/TouchpointTest.java b/bundles/org.eclipse.equinox.p2.tests/src/org/eclipse/equinox/p2/tests/engine/TouchpointTest.java
index a814cc1..6c7e078 100644
--- a/bundles/org.eclipse.equinox.p2.tests/src/org/eclipse/equinox/p2/tests/engine/TouchpointTest.java
+++ b/bundles/org.eclipse.equinox.p2.tests/src/org/eclipse/equinox/p2/tests/engine/TouchpointTest.java
@@ -108,7 +108,7 @@ public class TouchpointTest extends AbstractProvisioningTest {
 	}
 
 	private ServiceReference engineRef;
-	private Engine engine;
+	private IEngine engine;
 
 	public TouchpointTest(String name) {
 		super(name);
@@ -119,8 +119,8 @@ public class TouchpointTest extends AbstractProvisioningTest {
 	}
 
 	protected void setUp() throws Exception {
-		engineRef = TestActivator.getContext().getServiceReference(Engine.class.getName());
-		engine = (Engine) TestActivator.getContext().getService(engineRef);
+		engineRef = TestActivator.getContext().getServiceReference(IEngine.SERVICE_NAME);
+		engine = (IEngine) TestActivator.getContext().getService(engineRef);
 	}
 
 	protected void tearDown() throws Exception {
